src/app/shared/input/input.component.html

```diff
@@ -0,0 +1,9 @@
+<div>
+  <span *ngIf="propName" (click)="isActiveInput = !isActiveInput">{{propName}}: </span>
+  <span *ngIf="!isActiveInput"
+        [ngClass]="{'changed-value': compareValue(previousValue, state), 'default-value': !compareValue(previousValue, state) && propName}"
+        (click)="isActiveInput = !isActiveInput">{{state}}
+  </span>
+  <input type="text" *ngIf="isActiveInput" (blur)="onBlur(state)" [(ngModel)]="state">
+</div>
```

Comment:
Instead of additional <div> wrapper you can use :host in your SCSS
:host {
   display: block;
}
```suggestion
  <span *ngIf="propName" (click)="isActiveInput = !isActiveInput">{{propName}}: </span>
  <span *ngIf="!isActiveInput"
        [ngClass]="{'changed-value': compareValue(previousValue, state), 'default-value': !compareValue(previousValue, state) && propName}"
        (click)="isActiveInput = !isActiveInput">{{state}}
  </span>
  <input type="text" *ngIf="isActiveInput" (blur)="onBlur(state)" [(ngModel)]="state">
```